src/routes/resource.js

```diff
@@ -107,5 +114,57 @@ module.exports = function resource (endpoint, exclude, associations) {
     });
   });
 
+  // Transaction method that links all the associated models in the body
+  // for an specific model. It returns a promise that resolves with the linked
+  // instance
+  const linkAssociations = (models, req, transaction, instance) => {
+    return new Promise((resolve, reject) => {
+      if (associations.length <= 0) {
+        // Nothing to link
+        return resolve(instance);
+      }
+
+      let promises = [];
+      associations.forEach(name => {
+        if (!req.body[name]) {
+          return;
+        }
+        const body = req.body[name];
+        const link = linkInstance(instance, models, name, body, transaction);
+        promises.push(link);
+      });
+
+      Promise.all(promises).then(() => {
+        resolve(instance);
+      }).catch(err => {
+        reject(err);
+      });
+    });
+  };
+
+  // Finds or create an specific entity, and associate it to the given instance.
+  const linkInstance = (instance, models, name, entity, transaction) => {
+    const model = models[name];
+    const id = entity[CONST.iotId] || -1;
```

Comment:
-1 is fine as long as the id field that Sequelize generates is not unsigned (I haven't checked). How does `findOrCreate` behaves with  `where: {}` or `where: { id: undefined }`? A test for this case would be great.